src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -1697,6 +1708,7 @@ void MPS::initialize_component_internal(const reg_t &qubits,
 }
 
 void MPS::reset(const reg_t &qubits, RngEngine &rng) {
+  move_all_qubits_to_sorted_ordering();
```

Comment:
As mentioned in #1122, performance can be improved by replacing this line with another logic

Reply:
I remember, but didn't want to do that in this PR. Having the qubits sorted simplifies the code.